File: app/src/main/java/com/example/shiranpeer/burritolocator/presenter/MainActivityRepository.java
package com.example.shiranpeer.burritolocator.presenter;

import com.example.shiranpeer.burritolocator.PlacesResult;

import io.reactivex.Observable;


public interface MainActivityRepository {
    Observable<PlacesResult> getNearbyPlaces(String restaurant, String s, String burrito, String distance, String googlePlaceApiKey);

    Observable<PlacesResult> getNearbyPlacesAdditionalResults(String pageToken, String googlePlaceApiKey);
}

File: app/src/main/java/com/example/shiranpeer/burritolocator/repository/MainActivityRepository.kt
package com.example.shiranpeer.burritolocator.repository

import com.example.shiranpeer.burritolocator.model.PlacesResult

import io.reactivex.Observable


interface MainActivityRepository {
    fun getNearbyPlaces(restaurant: String, s: String, burrito: String, distance: String, googlePlaceApiKey: String): Observable<PlacesResult>?

    fun getNearbyPlacesAdditionalResults(pageToken: String, googlePlaceApiKey: String): Observable<PlacesResult>
}
